package/MDAnalysis/lib/distances.py

```diff
@@ -367,75 +362,82 @@ def self_distance_array(reference, box=None, result=None, backend="serial"):
+        triclinic and must be provided in the same format as returned by
+        :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:\n
+        ``[lx, ly, lz, alpha, beta, gamma]``.
+    method : str, optional
+        Keyword to override the automatic guessing ofthe employed search method.
+        Can be ``'bruteforce'``, ``'nsgrid'``,  or ``'pkdtree'``.
+    return_distances : bool, optional
+        If set to ``True``, distances will also be returned.
 
     Returns
     -------
-    pairs : array
-        Pair of indices, one from each reference and configuration such that
-        distance between them is  within the ``max_cutoff`` and ``min_cutoff``
-        pairs[i,j] contains the indices i from reference coordinates, and
-        j from configuration
-    distances : (optional) array
-        Distances corresponding to each pair of indices.
-        d[k] corresponding to the pairs[i,j] gives the distance between
-        i-th and j-th coordinate in reference and configuration respectively
-        Returns only if ``return_distances`` is set to `True`
+    pairs : numpy.ndarray
+        Pairs of indices, corresponding to coordinates in the `reference` and
+        `configuration` arrays such that the distance between them lies within
+        the interval (`min_cutoff`, `max_cutoff`].
+        Each row in `pairs` is an index pair ``[i, j]`` corresponding to the
+        ``i``-th coordinate in `reference` and the ``j``-th coordinate in
+        `configuration`.
+    distances : numpy.ndarray, optional
+        Distances corresponding to each pair of indices. Only returned if
+        `return_distances` is ``True``. ``distances[k]`` corresponds to the
+        ``k``-th pair returned in `pairs` and gives the distance between the
+        coordinates ``reference[pairs[k, 0]]`` and
+        ``configuration[pairs[k, 1]]``.
 
         .. code-block:: python
 
-            pairs, distances = capped_distances(reference, coordinates, max_cutoff)
-            for indx, [a,b] in enumerate(pairs):
-                coord1 = reference[a]
-                coord2 = configuration[b]
-                distance = distances[indx]
+            pairs, distances = capped_distances(reference, configuration,
+                                                max_cutoff, return_distances=True)
+            for k, [i, j] in enumerate(pairs):
+                coord1 = reference[i]
+                coord2 = configuration[j]
+                distance = distances[k]
 
     Note
     -----
-    Currently only supports brute force and Periodic KDtree
+    Currently supports brute force, grid-based, and periodic KDtree search
+    methods.
 
-    .. SeeAlso:: :func:'MDAnalysis.lib.distances.distance_array'
-    .. SeeAlso:: :func:'MDAnalysis.lib.pkdtree.PeriodicKDTree.search'
-    .. SeeAlso:: :class:'MDAnalysis.lib.nsgrid.FastNS.search'
+    .. seealso:: :meth:`~MDAnalysis.lib.distances.distance_array`
```

Comment:
This should use NUmpyDoc 
```reST
See Also
--------
MDAnalysis.lib.distances.distance_array
MDAnalysis.lib.pkdtree.PeriodicKDTree.search
MDAnalysis.lib.nsgrid.FastNS.search
```
see https://numpydoc.readthedocs.io/en/latest/format.html#sections, _11. See Also_

Reply:
Right, will do.